Draft: Pending rename of package py-dockerpy-creds [py-wandb] added version 0.12.10

qwertos opened the pull request.

qwertos committed:
[py-wandb] switched to pypi and audited dependencies

qwertos committed:
[py-wandb] added version 0.12.10

qwertos committed:
[py-wandb] flake8

adamjstewart commented on var/spack/repos/builtin/packages/py-wandb/package.py:
```diff
@@ -11,23 +11,33 @@ class PyWandb(PythonPackage):
+    depends_on('py-psutil@5.0:',            type=('build', 'run'))
+    depends_on('py-sentry-sdk@0.4:',        type=('build', 'run'))
+    depends_on('py-sentry-sdk@1:',          type=('build', 'run'), when='@0.12.10:')
+    depends_on('py-subprocess32@3.5.3:',    type=('build', 'run'))
```
This isn't required in the latest version

adamjstewart commented on var/spack/repos/builtin/packages/py-wandb/package.py:
```diff
@@ -11,23 +11,33 @@ class PyWandb(PythonPackage):
+    depends_on('py-sentry-sdk@0.4:',        type=('build', 'run'))
+    depends_on('py-sentry-sdk@1:',          type=('build', 'run'), when='@0.12.10:')
+    depends_on('py-subprocess32@3.5.3:',    type=('build', 'run'))
+    depends_on('py-dockerpy-creds@0.4:',    type=('build', 'run'))
```
```suggestion
    depends_on('py-docker-pycreds@0.4:',    type=('build', 'run'))
```
This package should be renamed to match PyPI.

adamjstewart commented on var/spack/repos/builtin/packages/py-wandb/package.py:
```diff
@@ -11,23 +11,33 @@ class PyWandb(PythonPackage):
-    depends_on('py-subprocess32', type=('build', 'run'))
-    depends_on('py-dockerpy-creds', type=('build', 'run'))
-    depends_on('py-configparser', type=('build', 'run'))
+    pypi     = "wandb/wandb-0.10.1.tar.gz"
```
```suggestion
    pypi     = "wandb/wandb-0.12.10.tar.gz"
```

qwertos committed:
[py-wandb]

- updated pypi variable to a version that is not deprecated
- py-subprocess32 not needed for newer version

adamjstewart commented:
Ping @qwertos

alalazo commented:
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.
